### ISSUE TITLE
Recover pre-training time from a fine-tuned checkpoint

### ISSUE THREAD

**Issue by moinnadeem** [mosaicml/composer#217]
## 🚀 Feature Request
We're in the following situation:
1. We pre-train some BERT model, and checkpoint it
2. We load a checkpoint and fine-tune it

However, we now need to create pareto plots based on these fine-tuned runs from pre-trained checkpoints. However, the fine-tuned run has no idea how long the pre-trained checkpoint was trained for, which makes it hard to generate our X axis, since the loaded checkpoint doesn't have access to the pretrained `wall_clock_time`.

Some suggestions:
1. We could expand the CheckpointLoader with a `restore_keys` parameter, that describes what part of the State to recover. Then, we can add `current_runtime` to the State, and recover that in the checkpoint. We can upload `pretrained_runtime` as a field in the Weights and Biases run, which makes generating the pareto plot easy.
2. We could write a custom callback, called something like a `PretrainedInformation` callback that can get recovered by the `restore_keys` parameter in checkpointing. This callback would contain the `pretrained_runtime` and manage uploading that into WandB on its own.
3. This is very hack-y, but we could also just name the file `{iterations}_{iterations_format}_{wall-clock-time}.pt`. This is easy to recover from the finetuned checkpoint, since it'll be reported in the `load_checkpoint.filepath` parameter, and a simple string split will recover the `{wall-clock-time}` parameter. It's pretty poor engineering practice though

Curious for thoughts from others. I might be able to take on the implementation if I have any free cycles, but I will likely be spread thin over the next few weeks, so a hand here could be appreciated.

I'm just going to implement #3 to unblock myself, and then we can operationalize which one we think is right.

**Comment by ravi-mosaicml** [comment]
This makes sense.

We need to track wall clock time. See #256.
Then, once #256 is implemented, the wall clock time can be parsed via something like the following:

```python
checkpoint_state_dict = torch.load("mosaic_states.pt")
timer = Timer()
timer.load_state_dict(checkpoint_state_dict["state"]["timer"])
elapsed_training_duration = timer.get_elapsed_training_duration()
```

@moinnadeem would this work?

**Comment by moinnadeem** [comment]
Yes! This will work.

It's important to note that I've already implemented #3, so we can deprioritize this if we want. I would like to have an engineering sprint in February where we work on fixing all of the hacks that I wrote to get BERT merged. However, if you have the cycles, then feel free to implement your current solution.

**Comment by moinnadeem** [comment]
@hanlint also mentioned wondering if this is feature that should be included in Composer or not. I wanted to check Ravi's proposed solution against him

**Comment by ravi-mosaicml** [comment]
Closing for #256
